### PR TITLE
METADATA: Adding generic meta support for oracle

### PR DESCRIPTION
**Pull request by asketagarwal**
METADATA: Adding generic meta support for oracle

### CODE REVIEW

File: lib/rucio/core/did.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 import logging
 import random
 import sys
+import json
```
Comment (mlassnig) [inline lib/rucio/core/did.py]: sort list of imports

File: lib/rucio/db/sqla/types.py [hunk 2]
```diff
@@ -113,5 +113,7 @@ def load_dialect_impl(self, dialect):
             return dialect.type_descriptor(JSONB())
         elif dialect.name == 'mysql':
             return dialect.type_descriptor(types.JSON())
+        elif dialect.name == 'oracle':
+        	return dialect.type_descriptor(CLOB())
```
Comment (mlassnig) [inline lib/rucio/db/sqla/types.py]: lib/rucio/db/sqla/types.py:117:9: E101 indentation contains mixed spaces and tabs
lib/rucio/db/sqla/types.py:117:9: W191 indentation contains tabs

**Comment by mlassnig** [comment]
for my understanding, this works by just storing the JSON as a clob in oracle? and we would need oracle12 working to get full JSON support, right?

**Comment by asketagarwal** [comment]
This does not work yet, when you see the oracle test you see it fails as it can't create the table. Oracle 12 is needed for that. Basically for json in oracle there needs to be a constraint on the clob type with IS JSON keyword which is not there in oracle 11

**Review by vingar — APPROVED** [review]

**Review by mlassnig — APPROVED** [review]

File: lib/rucio/core/did.py [hunk 3]
```diff
@@ -1285,12 +1286,17 @@ def get_did_meta(scope, name, session=None):
     :param scope: the scope of did
     :param name: the name of the did
     """
-    if session.bind.dialect.name == 'oracle' or session.bind.dialect.name == 'sqlite':
+    if session.bind.dialect.name == 'sqlite':
         raise NotImplementedError
+    if session.bind.dialect.name == 'oracle':
```
Comment (bari12) [inline lib/rucio/core/did.py:1291]: Just an idea for later: why might think about doing these db compatibility checks via a function decorator?

**Pull request merged**